app/main.py

```diff
@@ -1,6 +1,28 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        key = (args, frozenset(kwargs.items()))
+        if key in cache_dict:
+            print("Getting from cache")
+            return cache_dict[key]
+        else:
+            print("Calculating new result")
+            cache_dict[key] = func(*args, **kwargs)
+            return cache_dict[key]
+    return wrapper
+
+
+# I wrote a new function because I deleted the old one
```

Comment:
You don't need these functions here. Remove tests from the main file, you already have tests for the decorator, they are in the different module. Don't mix tests and main logic